source/darwin/secure_transport_tls_channel_handler.c

```diff
@@ -612,7 +612,8 @@ static int s_process_read_message(
         struct aws_io_message *outgoing_read_message = aws_channel_acquire_message_from_pool(
             slot->channel, AWS_IO_MESSAGE_APPLICATION_DATA, downstream_window - processed);
         if (!outgoing_read_message) {
-            return AWS_OP_ERR;
+            /* even though this is a failure, this handler has taken ownership of the message */
```

Comment:
don't you need to shutdown here as well? this assumes an allocation failed.

Reply:
Sure, for now. I'll add it, but this should crash and die before ever returning here.